shell-control/agent/backend/shellcontrol.py

```
@@ -0,0 +1,181 @@
+        return self.client.get_usage_summary()
+
+    async def call_tool(self, parameters: dict, function, formatter_function):
+        try:
+            tool_result = await function(**parameters)
+            if formatter_function == None:
+                self.client.append_user_message(tool_result)
+            else:
+                tool_result_formatted = formatter_function(tool_result)
+                self.client.append_user_message(tool_result_formatted)
+            return self.event(Event.TOOL_SUCCESS, tool_result)
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)